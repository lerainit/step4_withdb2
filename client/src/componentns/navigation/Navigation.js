import React from 'react'
import { NavLink } from 'react-router-dom'
import styles from './Navigation.module.scss'
import { useSelector } from 'react-redux'
import BackButton from '../backbutton/backbutton'


const Navigation = () => {

const users= useSelector(store => store.users.value)
    

    return (
       
    <header className={styles.header}>
       
            <NavLink className={styles.logo} to='/' ><h2 >Instagram</h2 ></NavLink><BackButton/>
                      <div className={styles.container}>
              {    users.map(({url,name,id},index) => <NavLink className={styles.cart_link} to ={`/user${id}`} ><img className={styles.cart} src={url} alt="img" /><h2 className={styles.user_name}>{name}</h2></NavLink>)
               
}

            </div></header>

)


}
export default Navigation