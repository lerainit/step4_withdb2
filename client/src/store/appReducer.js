import { combineReducers } from "redux";
import cardIndexReducer from "./cardId/reducer";
import modalReducer from './modal/reducer'
import productsReducer from "./products/reducer";
import likesReducer from "./likes/reducer";
import BackgroundReducer from "./cardBackground/reducer";
import postsReducer from "./posts/reducer";
import UsersReducer from "./users/reducer";
import subscriberReducer from "./subscribers/reducer";
import userIndexReducer from "./userIndex/reducer";
import commentsReducer from "./comments/reducer";
import FollowerReducer from "./followers/reducer";


const appReducer = combineReducers({
   modal: modalReducer,
   cardIndex: cardIndexReducer,
   products: productsReducer,
   counter: likesReducer,
   background: BackgroundReducer,
   posts: postsReducer,
   users: UsersReducer,
   subscribers: subscriberReducer,
   userIndex: userIndexReducer,
   comments: commentsReducer,
   followers:FollowerReducer

})
export default appReducer