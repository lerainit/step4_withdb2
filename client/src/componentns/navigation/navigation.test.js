import { render, screen } from '@testing-library/react';
import Navigation from './Navigation';

import { Provider } from "react-redux";
import store from '../../store'
import { BrowserRouter } from 'react-router-dom';

const Component =() =>{
    return (
    <Provider store ={store}>
        <BrowserRouter>
        <Navigation/>
        </BrowserRouter>
    </Provider>
)}
describe('Navigation renders',()=>{

test('should Navigation match snapshot',()=>{
const {asFragment } = render(<Component/>)
expect(asFragment()).toMatchSnapshot()

})

})