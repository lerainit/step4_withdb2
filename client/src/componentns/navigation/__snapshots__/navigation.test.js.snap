// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation renders should Navigation match snapshot 1`] = `
<DocumentFragment>
  <header
    class="header"
  >
    <a
      aria-current="page"
      class="logo active"
      href="/"
    >
      <h2>
        Instagram
      </h2>
    </a>
    <div
      class="container"
    />
  </header>
</DocumentFragment>
`;
